app/javascript/controllers/count_controller.js

```diff
@@ -0,0 +1,25 @@
+import { Controller } from "@hotwired/stimulus"
+
+export default class extends Controller {
+  static targets = ["counter", "display"]
+  static values = {
+    selected: { type: Number, default: 0 },
```

Comment:
Is this even being passed in anywhere, or are you getting the default?

I'd probably suggest omitting this entirely.  The server should render the correct number in the HTML to start with.  And then drop the `connect()` method here, and just wait until some action calls `recount()` to have this controller modify the innerHTML.